pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -564,6 +576,16 @@ func (gce *GCECloud) HasClusterID() bool {
 	return true
 }
 
+// Project IDs cannot have a digit for the first characeter. If the id contains a digit,
+// then it must be a project number.
+func isProjectNumber(idOrNumber string) bool {
+	if len(idOrNumber) == 0 {
+		return false
+	}
+
+	return idOrNumber[0] >= '0' && idOrNumber[0] <= '9'
```

Comment:
Shouldn't this actually parse to see it's a number? Why do we want to make the assumption it is in a particular format?

Reply:
My original code from several months ago tried parsing it, but you suggested to go this route.

Reply:
On second look, I was wrong, probably should parse it as a number, unless it's too big?